cmd/channel/webhook/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/icinga/icinga-notifications/internal"
+	"github.com/icinga/icinga-notifications/pkg/plugin"
+	"io"
+	"net/http"
+	"text/template"
+)
+
+type Webhook struct {
+	Method              string `json:"method"`
+	URLTemplate         string `json:"url_template"`
+	RequestBodyTemplate string `json:"request_body_template"`
+
+	tmplUrl         *template.Template
+	tmplRequestBody *template.Template
+}
+
+func (ch *Webhook) GetInfo() *plugin.Info {
+	elements := []*plugin.ConfigOption{
+		{
+			Name: "method",
+			Type: "option",
+			Label: map[string]string{
+				"en_US": "HTTP Method",
+				"de_DE": "HTTP-Methode",
+			},
+			Help: map[string]string{
+				"en_US": "HTTP request method used for the web request.",
+				"de_DE": "HTTP-Methode für die Anfrage.",
+			},
+			Default:  "POST",
+			Required: true,
+			Options: map[string]string{
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+				"GET":     "GET",
+				"HEAD":    "HEAD",
+				"POST":    "POST",
+				"PUT":     "PUT",
+				"DELETE":  "DELETE",
+				"CONNECT": "CONNECT",
+				"OPTIONS": "OPTIONS",
+				"TRACE":   "TRACE",
+				"PATCH":   "PATCH",
+			},
```

Comment:
That's an interesting list for sure. I guess the idea behind providing all of them was so that there are no arbitrary restrictions? But on the other hand, this will now provide quite some strange suggestions.

Another option could be to just use an arbitrary text input with default `POST`, that will probably be the value of choice in almost all cases anyways.

If there was a way to express "or a custom value" with `Type: "option"`, I think that would be the obvious choice, but at the moment, that's not possible. So I'm not entirely sure what the best option is at the moment.

Reply:
I guess this boils down to the question if this channel should cover HTTP in all its glory and with all its features or to cover _just enough_, which will eventually also lead to feature creep in the long run.

When I first started this, I did enough for the integration tests with some additional knobs. But, as I also wrote in my [other comment](https://github.com/Icinga/icinga-notifications/pull/149#discussion_r1568933901), I don't think one can develop a maintainable channel for everything which might be covered under the name _webhook_.

On the one hand, I would argue to limit this channel to the integration test's necessaries and later move it into a testing sub-directory. However, on the other hand, real users are going to request something HTTP-speaking as most external notification/messaging services are having an HTTP API.